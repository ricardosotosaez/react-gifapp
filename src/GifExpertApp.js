import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {


    const [categories, setCategories] = useState(['The Beatles']);

    return(
        <>        
            <h1 className='titulo animate__animated animate__tada'>GIF APP</h1>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {    
                    categories.map(category => (
                       <GifGrid 
                        key={category}    
                        category={category}
                       />
                    ))
                }
            </ol>
        </>
    );
};

export default GifExpertApp;
