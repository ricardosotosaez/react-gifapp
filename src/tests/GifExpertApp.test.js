import React from 'react';
import'@testing-library/jest-dom';
import {shallow} from 'enzyme';

import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {

    test('debe de cargar correctamente', () => {
        const wrapper =  shallow(<GifExpertApp />)        
        expect(wrapper).toMatchSnapshot();
    })

})
