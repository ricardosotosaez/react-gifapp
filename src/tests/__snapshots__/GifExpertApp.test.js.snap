// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de cargar correctamente 1`] = `
<Fragment>
  <h1
    className="titulo animate__animated animate__tada"
  >
    GIF APP
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The Beatles"
      key="The Beatles"
    />
  </ol>
</Fragment>
`;
